src/vm/ceemain.cpp

```diff
@@ -1315,8 +1315,7 @@ void STDMETHODCALLTYPE EEShutDownHelper(BOOL fIsDllUnloading)
     }
 
 #ifdef FEATURE_PERFTRACING
-    // Shutdown the event pipe.
-    EventPipe::Shutdown();
+    // Shutdown the diagnostics server first.
     DiagnosticServer::Shutdown();
```

Comment:
Why do we want to shutdown the diagnostic server early?

Reply:
Terminating the server first before stopping EventPipe seems reasonable to me. It would be weird to have to process a request trying to enable logging after EventPipe has already shutdown.

Reply:
Actually, now that I think about it, we should shut this down after EventPipe finished shutting down because we might want to still send some stuff over through the IPC channel which gets freed during this call. Thanks for the catch Jan.